src/redux/network/index.ts

```diff
@@ -0,0 +1,47 @@
+import { Action } from 'redux'
+
+export enum ActionTypes {
+  NETWORK_SET_VALID_CHAIN_ID = 'NETWORK_SET_VALID_CHAIN_ID',
+  NETWORK_SET_INVALID_CHAIN_ID = 'NETWORK_SET_INVALID_CHAIN_ID',
+}
+
+interface NetworkActionSetValidChainId extends Action<ActionTypes.NETWORK_SET_VALID_CHAIN_ID> {
+  payload: number
+}
+
+interface NetworkActionSetInvalidChainId extends Action<ActionTypes.NETWORK_SET_INVALID_CHAIN_ID> {
+  payload: number
+}
+
+export type NetworkAction = NetworkActionSetValidChainId | NetworkActionSetInvalidChainId
+
+export type NetworkState = {
+  validChainId: number | null
+  invalidChainId: number | null
+}
```

Comment:
We can just use `validChainId` as a boolean flag. `invalidChainId` seems like an overkill.

Reply:
Sure, I had thought it might be useful later, but you might be right